src/test/pmemblk_create/pmemblk_create.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2015, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * pmemblk_create.c -- unit test for creating a block memory pool
+ *
+ * usage: pmemblk_create path bsize
+ */
+
+#include "unittest.h"
+
+#define	SIZEOF_TESTFILE	(64 * 1024 * 1024)
+#define	CREATE_MODE		(0664)
+#define	CHECK_BYTES		(4096)
+
+int
+main(int argc, char *argv[])
+{
+	START(argc, argv, "pmemblk_create");
+
+	if (argc != 3)
+		FATAL("usage: %s path bsize", argv[0]);
+
+	PMEMblkpool *handle;
+	const char *path = argv[1];
+	size_t bsize = strtoul(argv[2], NULL, 0);
+
+	int fd;
+
+	if (strcmp(path, "NULLFILE") == 0) {
+
+		if ((handle = pmemblk_create("./testfile", bsize,
```

Comment:
A name of a test file should not be hard coded but given as an argument (see src/test/pmem_is_pmem/pmem_is_pmem.c:49 for example), because its path is set in the config file src/test/testconfig.sh and it does not have to be located in the current test directory. There are more occurrences of the same issue listed below:
./src/test/pmemblk_create/pmemblk_create.c:63:			OUT("!./testfile: pmemblk_create");
./src/test/pmemblk_create/pmemblk_create.c:67:		int result = pmemblk_check("./testfile");
./src/test/pmemblk_create/pmemblk_create.c:70:			OUT("!%s: pmemblk_check", "./testfile");
/src/test/pmemblk_create/pmemblk_create.c:72:			OUT("%s: pmemblk_check: not consistent", "./testfile");
./src/test/pmemblk_create/pmemblk_create.c:74:		fd = OPEN("./testfile", O_RDWR);
./src/test/pmemblk_open/pmemblk_open.c:58:		if ((handle = pmemblk_create("./testfile", bsize_create,
./src/test/pmemblk_open/pmemblk_open.c:60:			OUT("!./testfile: pmemblk_create");
./src/test/pmemblk_open/pmemblk_open.c:64:		if ((handle = pmemblk_open("./testfile", bsize_open)) == NULL) {
./src/test/pmemblk_open/pmemblk_open.c:65:			OUT("!./testfile: pmemblk_open");
./src/test/pmemlog_create/pmemlog_create.c:56:		if ((handle = pmemlog_create("./testfile",
./src/test/pmemlog_create/pmemlog_create.c:58:			OUT("!./testfile: pmemlog_create");
./src/test/pmemlog_open/pmemlog_open.c:56:		if ((handle = pmemlog_create("./testfile",
./src/test/pmemlog_open/pmemlog_open.c:58:			OUT("!./testfile: pmemlog_create");
./src/test/pmemlog_open/pmemlog_open.c:62:		if ((handle = pmemlog_open("./testfile")) == NULL) {
./src/test/pmemlog_open/pmemlog_open.c:63:			OUT("!./testfile: pmemlog_open");